src/rgw/rgw_sal_motr.cc

```diff
@@ -2228,14 +2247,12 @@ int MotrObject::remove_mobj_and_index_entry(
         if (rc < 0) {
           ldpp_dout(dpp, 0) <<__func__<< ": ERROR: get_upload_id failed. rc=" << rc << dendl;
         } else {
-          std::string obj_fqdn = bucket_name + "/" + delete_key;
-          std::string obj_part_iname = "motr.rgw.object." + bucket_name + "." +
-                                       this->get_name() + "." + upload_id +
-                                       ".parts";
-          ldpp_dout(dpp, 20) << __func__ << ": object part index=" << obj_part_iname << dendl;
+          std::string obj_fqdn = delete_key + "." + upload_id;
```

Comment:
While writing parts we don't have version_id generated, it is generated in the complete stage.
thus here delete_key will be  `obj_name + "\a" + version_id` (id is generated), 
but in upload part i.e. MultipartWriter::complete  key will be   `obj_name`.
```suggestion
          std::string obj_fqdn = this->get_name() + "." + upload_id;
```

Reply:
**JFYI**
Now @sachinpunadikar , @jjxsg the meaning/use of ginfo.name is different for multipart and simple obj.
**for multipart:**  we are using it to search obj parts in the bucket_parts_index_table and it does have a bucket name in it
**for simple:**  we are not using it at all, it is only for information and it contains the bucket name.

Reply:
@sumedhak27 `ginfo.name` is important for listing GC entries irrespective of type of object.

Reply:
@sumedhak27, good catch. Thanks!